### ISSUE TITLE
Integration DeepSource

### ISSUE THREAD

**Issue by ankitdobhal** [pingcap/tidb#23827]
## Description
 
Hi 👋 I ran the [DeepSource](https://deepsource.io/) static analyzer on the forked copy of this repo and found some interesting code quality issues which are available [here](https://deepsource.io/gh/ankitdobhal/tidb/issues/).
I opened a PR (#23445 ) fixing some of the issues detected for you to assess if it is right for you.

**Describe the feature you'd like:**

The Static Code Analysis Tool found potential bugs and anti-patterns in the Code, that can be detrimental at a later point in time with respect to the Project. DeepSource helps you to automatically find and fix issues in your code during code reviews. This tool looks for anti-patterns, bug risks, performance problems, and raises issues.

There are plenty of other issues in relation to Bug Discovery and Anti-Patterns which you would be interested to take a look at.
If you would like to integrate DeepSource to autofix some of the common occurring issues, I can help you set that up :)

**Comment by zz-jason** [comment]
Good catch! I removed the 'feature-request' label since it's more about code quality.
